lib/naughty/core.lua

```diff
@@ -583,6 +583,7 @@ function naughty.notify(args)
             die(naughty.notificationClosedReason.dismissedByUser)
         end
     end
+    notification.run = run
```

Comment:
Uhm, why? Both of these could be done by the caller. Instead of `local foo = naughty.run{ actions = actions, run = run }`, just do `local foo = { actions = actions, run = run, notification = naughty.run{ actions = actions, run = run }}` and fix up all surrounding code.
(Well, ok, if you do not use a run callback, you need to call `naughty.destroy(notification, naughty.notificationClosedReason.dismissedByUser)` directly. Still, I fail to see the point.)